extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -831,28 +900,44 @@ private String getDeobfuscationCode() throws ParsingException {
                 playerJsUrl = HTTPS + "//www.youtube.com" + playerJsUrl;
             }
 
-            cachedDeobfuscationCode = loadDeobfuscationCode(playerJsUrl);
+            final String deobfuscationCode = loadDeobfuscationCode(playerJsUrl);
+            final String swapFirstAndIndexName = Parser.matchGroup1(SWAP_FIRST_AND_INDEX_REGEX, deobfuscationCode);
+            final String sliceAtName = Parser.matchGroup1(SLICE_AT_REGEX, deobfuscationCode);
+            final String reverseName = Parser.matchGroup1(REVERSE_REGEX, deobfuscationCode);
+            final String containerName = Parser.matchGroup1(CONTAINER_REGEX, deobfuscationCode);
+
+            final String[] stringSteps = deobfuscationCode.split(containerName + "\\.");
+            boolean first = true;
+            cachedDeobfuscationFunctions = new ArrayList<>();
+            for (final String step : stringSteps) {
```

Comment:
Since this is an array you can just use a `for i` loop with `i=1` instead of the weird `first=true` setup

Reply:
or `cachedDeobfuscationFunctions.remove(0);`